command/job_scaling_events.go

```diff
@@ -0,0 +1,189 @@
+	return "Display the most recent scaling events for a job"
+}
+
+func (j *JobScalingEventsCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(j.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"-verbose": complete.PredictNothing,
+		})
+}
+
+// Name returns the name of this command.
+func (j *JobScalingEventsCommand) Name() string { return "job scaling-events" }
+
+// Run satisfies the cli.Command Run function.
+func (j *JobScalingEventsCommand) Run(args []string) int {
+
+	var verbose bool
+
+	flags := j.Meta.FlagSet(j.Name(), FlagSetClient)
+	flags.Usage = func() { j.Ui.Output(j.Help()) }
+	flags.BoolVar(&verbose, "verbose", false, "")
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
+	args = flags.Args()
+	if len(args) != 1 {
+		j.Ui.Error("This command takes one argument: <job_id>")
+		j.Ui.Error(commandErrorText(j))
+		return 1
+	}
+
+	// Get the job ID.
+	jobID := args[0]
+
+	// Get the HTTP client.
+	client, err := j.Meta.Client()
+	if err != nil {
+		j.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
+		return 1
+	}
+
+	events, _, err := client.Jobs().ScaleStatus(jobID, nil)
+	if err != nil {
+		j.Ui.Error(fmt.Sprintf("Error listing scaling events: %s", err))
+		return 1
+	}
+
+	// Check if any of the task groups have scaling events, otherwise exit
+	// indicating there are not any.
+	var haveEvents bool
+	for _, tg := range events.TaskGroups {
+		if tg.Events != nil {
+			haveEvents = true
+			break
+		}
+	}
+
+	if !haveEvents {
+		j.Ui.Output("No events found")
+		return 0
+	}
+
+	// Create our sorted list of events and output.
+	sortedList := sortedScalingEventList(events)
+	j.Ui.Output(formatList(formatScalingEventListOutput(sortedList, verbose, 0)))
+	return 0
+}
+
+func formatScalingEventListOutput(e scalingEventList, verbose bool, limit int) []string {
+
+	// If the limit is zero, aka no limit or the limit is greater than the
+	// number of events we have then set it this to the length of the event
+	// list.
+	if limit == 0 || limit > len(e) {
+		limit = len(e)
+	}
+
+	// Create the initial output heading.
+	output := make([]string, limit+1)
+	output[0] = "Task Group|Count|PrevCount"
+
+	// If we are outputting verbose information, add these fields to the header
+	// and then add our end date field.
+	if verbose {
+		output[0] += "|Error|Message|Eval ID"
+	}
+	output[0] += "|Date"
+
+	var i int
+
+	for i < limit {
+		output[i+1] = fmt.Sprintf("%s|%v|%v", e[i].name, *e[i].event.Count, e[i].event.PreviousCount)
+		if verbose {
+			output[i+1] += fmt.Sprintf("|%v|%s|%s",
+				e[i].event.Error, e[i].event.Message, *e[i].event.EvalID)
+		}
+		output[i+1] += fmt.Sprintf("|%v", formatTime(time.Unix(0, int64(e[i].event.Time))))
+		i++
+	}
+	return output
+}
+
+// sortedScalingEventList generates a time sorted list of scaling events as
+// provided by the api.JobScaleStatusResponse.
+func sortedScalingEventList(e *api.JobScaleStatusResponse) []groupEvent {
```

Comment:
if scaling events aren't already sorted by time, it's a bug. 
https://github.com/hashicorp/nomad/blob/master/nomad/state/state_store.go#L669-L673

Reply:
Ah, now I remember why I did this. It allows us to order the scaling events across the whole job, rather than having them ordered within the confines of each task group within the job. I think this is useful, particularly when a job contains several task groups all with scaling events. Thoughts?

Reply:
ah, yeah, that makes sense. thanks!